mirrord-agent/src/file.rs

```diff
@@ -288,6 +297,32 @@ impl FileManager {
             })
     }
 
+    #[tracing::instrument(level = "trace", skip(self))]
+    pub(crate) fn write_limited(
+        &mut self,
+        fd: usize,
+        start_from: u64,
+        buffer: Vec<u8>,
+    ) -> RemoteResult<WriteFileResponse> {
+        self.open_files
+            .get_mut(&fd)
+            .ok_or(ResponseError::NotFound(fd))
+            .and_then(|remote_file| {
+                if let RemoteFile::File(file) = remote_file {
+                    let mut writer = BufWriter::new(std::io::Write::by_ref(file));
+                    let _ = writer.seek(SeekFrom::Start(start_from))?;
```

Comment:
you need to seek to the original offset as `The pwrite() function is equivalent to write(), except that it writes into a given position and does not change the file offset (regardless of whether O_APPEND is set). The first three arguments to pwrite() are the same as write(), with the addition of a fourth argument offset for the desired position inside the file.`